server/src/test/java/org/opensearch/index/shard/IndexShardTests.java

```diff
@@ -327,7 +327,8 @@ public void testShardStateMetaHashCodeEquals() {
         ShardStateMetadata meta = new ShardStateMetadata(
             randomBoolean(),
             randomRealisticUnicodeOfCodepointLengthBetween(1, 10),
-            allocationId
+            allocationId,
+            randomFrom(ShardStateMetadata.IndexDataLocation.values())
         );
 
         assertEquals(meta, new ShardStateMetadata(meta.primary, meta.indexUUID, meta.allocationId));
```

Comment:
Please update to .. Tests are failing since we are picking random indexDataLocation and we are selecting Local as default in case of no DataLocation passed. This is failing the test case most of the times.
```
        assertEquals(meta, new ShardStateMetadata(meta.primary, meta.indexUUID, meta.allocationId, meta.indexDataLocation));
        assertEquals(meta.hashCode(), new ShardStateMetadata(meta.primary, meta.indexUUID, meta.allocationId, meta.indexDataLocation).hashCode());
```